---
layout: post
title: "Python 풀스택개발자 삽질로그 내컴 안에 뱀이 들어있다 000. 제발 파이썬 가상환경과 장고(Django) 좀 설치하게 해주세요"
categories: [Python, Development Environment]
tags: 
- Python
- Development Environment
---

<br>노마드코더 에어비앤비 클론강의대로라면 파이썬3과 VScode, 그리고 pipenv 순서대로 깔끔하게 세팅한 뒤 장고(Django)를 설치하면 되는 것이었다. 
그러나 내가 그대로 하면 자꾸 저런 식으로 나왔다.

~~~

(base) 192:~ spica$ cd Desktop
(base) 192:Desktop spica$ mkdir AirB && cd AirB
(base) 192:AirB spica$ pipenv --three
Creating a virtualenv for this project…
Pipfile: /Users/spica/Pipfile
Using /Users/spica/anaconda3/bin/python3 (3.7.3) to create virtualenv…
⠹ Creating virtual environment...Already using interpreter /Users/spica/anaconda3/bin/python3
Using base prefix '/Users/spica/anaconda3'
Running virtualenv with interpreter /Users/spica/anaconda3/bin/python3

✘ Failed creating virtual environment 
[pipenv.exceptions.VirtualenvCreationException]:   File "/Users/spica/anaconda3/lib/python3.7/site-packages/pipenv/vendor/click/decorators.py", line 17, in new_func
[pipenv.exceptions.VirtualenvCreationException]:       return f(get_current_context(), *args, **kwargs)
[pipenv.exceptions.VirtualenvCreationException]:   File "/Users/spica/anaconda3/lib/python3.7/site-packages/pipenv/cli/command.py", line 208, in cli
[pipenv.exceptions.VirtualenvCreationException]:       clear=state.clear,
[pipenv.exceptions.VirtualenvCreationException]:   File "/Users/spica/anaconda3/lib/python3.7/site-packages/pipenv/core.py", line 574, in ensure_project
[pipenv.exceptions.VirtualenvCreationException]:       pypi_mirror=pypi_mirror,
[pipenv.exceptions.VirtualenvCreationException]:   File "/Users/spica/anaconda3/lib/python3.7/site-packages/pipenv/core.py", line 506, in ensure_virtualenv
[pipenv.exceptions.VirtualenvCreationException]:       python=python, site_packages=site_packages, pypi_mirror=pypi_mirror
[pipenv.exceptions.VirtualenvCreationException]:   File "/Users/spica/anaconda3/lib/python3.7/site-packages/pipenv/core.py", line 935, in do_create_virtualenv
[pipenv.exceptions.VirtualenvCreationException]:       extra=[crayons.blue("{0}".format(c.err)),]
[pipenv.exceptions.VirtualenvCreationException]: Traceback (most recent call last):
  File "/Users/spica/anaconda3/lib/python3.7/site-packages/virtualenv.py", line 420, in copyfile
    os.symlink(os.path.realpath(src), dest)
FileExistsError: [Errno 17] File exists: '/Users/spica/anaconda3/lib/python3.7/lib-dynload' -> '/Users/spica/.local/share/virtualenvs/spica-E-kbpXJS/lib/python3.7/lib-dynload'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/spica/anaconda3/lib/python3.7/site-packages/virtualenv.py", line 2634, in <module>
    main()
  File "/Users/spica/anaconda3/lib/python3.7/site-packages/virtualenv.py", line 870, in main
    symlink=options.symlink,
  File "/Users/spica/anaconda3/lib/python3.7/site-packages/virtualenv.py", line 1162, in create_environment
    install_python(home_dir, lib_dir, inc_dir, bin_dir, site_packages=site_packages, clear=clear, symlink=symlink)
  File "/Users/spica/anaconda3/lib/python3.7/site-packages/virtualenv.py", line 1449, in install_python
    copy_required_files(stdlib_dir, lib_dir, symlink)
  File "/Users/spica/anaconda3/lib/python3.7/site-packages/virtualenv.py", line 1356, in copy_required_files
    copyfile(join(src_dir, fn), join(lib_dir, fn), symlink)
  File "/Users/spica/anaconda3/lib/python3.7/site-packages/virtualenv.py", line 423, in copyfile
    copy_file_or_folder(src, dest, symlink)
  File "/Users/spica/anaconda3/lib/python3.7/site-packages/virtualenv.py", line 401, in copy_file_or_folder
    shutil.copytree(src, dest, symlink)
  File "/Users/spica/anaconda3/lib/python3.7/shutil.py", line 321, in copytree
    os.makedirs(dst)
  File "/Users/spica/anaconda3/lib/python3.7/os.py", line 221, in makedirs
    mkdir(name, mode)
FileExistsError: [Errno 17] File exists: '/Users/spica/.local/share/virtualenvs/spica-E-kbpXJS/lib/python3.7/lib-dynload'

Failed to create virtual environment.

~~~

<br>니콜라스는 간단하게 설치되는 가상환경이 나는 왜 수십번을 해도 실패하는 것이며, 경로는 왜 Pipfile: /Users/spica/Pipfile 이라고 Desktop과 디렉토리인 AirB가 무시된 상태로 나오는 것인지... 의아한 점이 한 두개가 아니었다.

<br>그래서 자세히 강의를 돌려본 결과, 니콜라스와 내 pipenv 설치과정의 차이점이 뭔가했더니 'Using /Users/spica/anaconda3/bin/python3 (3.7.3) to create virtualenv…' 이 부분이 결정적으로 달랐다.

<br>아나콘다...
<br>아나콘다...
<br> **아나콘다?!!!** ~~네가 왜 거기서 나와??! (상상도 못한 정체)~~ 


